### PR TITLE
CFn Resource Provider Framework v2

### PR DESCRIPTION
**Pull request by dominikschubert**
## Changes

- Additional nesting for generated test files. `.../iam/test_aws_iam_user_basic.py` becomes `.../iam/aws_iam_user/test_basic.py`. This allows better segregation of resource types and simplifies file naming.
- Fixes for type generation based on the schema (e.g. for `AWS::ApiGateway::RestApi` and `AWS::DynamoDB::Table`)
- Introduce new config variable `CFN_RESOURCE_PROVIDER_OVERRIDES` to opt-in or opt-out of specific resource provider implementations.
- Remove previous internal `CFN_RESOURCE_PROVIDERS_V2` config variable. (This variable hasn't worked yet, so the removal doesn't affect any existing users)
- Migrate `AWS::IAM::User` tests to new structure

## Next up (Follow-up PR)
- We still need to fix the execution of new resource providers. More precisely, all the parts around the execution like depdency detection, GetAtt access, etc. which currently still tries to use the old GenericBaseModels for *all* resource types.

### CODE REVIEW

**Review by pinzon — APPROVED** [review]
LGTM

File: localstack/services/cloudformation/engine/template_deployer.py [hunk 1]
```diff
@@ -783,6 +783,15 @@ class TemplateDeployer:
     def __init__(self, stack):
         self.stack = stack
 
+        try:
+            self.provider_config = json.loads(config.CFN_RESOURCE_PROVIDER_OVERRIDES)
+        except json.JSONDecodeError:
+            LOG.warning(
+                "Failed to parse CFN_RESOURCE_PROVIDERS_OVERRIDE config. Not a valid JSON document.",
```
Comment (dominikschubert) [inline localstack/services/cloudformation/engine/template_deployer.py:790]:
```suggestion
                "Failed to parse CFN_RESOURCE_PROVIDER_OVERRIDES config. Not a valid JSON document.",
```

**Review by whummer — APPROVED** [review]
LGTM 👍 - looking forward to seeing the v2 provider framework in action 🚀

File: localstack/services/cloudformation/resource_provider.py [hunk 2]
```diff
@@ -639,22 +649,29 @@ def execute_action(
             case _:
                 raise NotImplementedError(change_type)
 
-    def load_resource_provider(self, resource_type: str) -> Optional[ResourceProvider]:
-        # TODO: unify behavior here in regards to raising NoResourceProvider
-        if not config.CFN_RESOURCE_PROVIDERS_V2:
+    def should_use_legacy_provider(self, resource_type: str) -> bool:
+        # any config overwrites take precedence over the default list
+        PROVIDER_CONFIG = {**PROVIDER_DEFAULTS, **self.provider_config}
```
[the remark this answers is not in the document]
Comment (dominikschubert) [reply localstack/services/cloudformation/resource_provider.py:654]: will address this in one of the follow-up PRs :+1:

**Pull request merged**